soc/intel/intel_adsp/ace/power.c

```diff
@@ -25,6 +25,14 @@
 #define SRAM_ALIAS_BASE         0xA0000000
 #define SRAM_ALIAS_MASK         0xF0000000
 
+#if CONFIG_SOC_INTEL_ACE15_MTPM
+/* Used to force any pending transaction by HW issuing an upstream read before
+ * power down host domain.
+ */
+uint32_t g_key_read_holder __aligned(CONFIG_DCACHE_LINE_SIZE);
+uint8_t g_key_read_holder_bottom_alignment[CONFIG_DCACHE_LINE_SIZE - sizeof(uint32_t)];
```

Comment:
Looks like this _bottom_alignment array is unused?   What's it for?   If the goal is to force padding for g_key_read_holder, that doesn't look like it's going to work.  We build with -fdata-sections and the linker will throw it out, leaving something arbitrary in that cache line.

Really just make the whole thing an aligned byte array of DCACHE_LINE_SIZE, you're going to cast it to a word pointer below anyway.

Reply:
indeed, thanks for pointing it out.